### PR TITLE
Disable Rule 0016 "Caption is missing." for objects of the the API

### PR DESCRIPTION
**Pull request by Arthurvdv**
Related to issue: https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/190.

Because this could be experienced as a breaking change I've provided a setting in the LinterCop.json to enable the current behavior. When an external application uses the metadata of the API's, like the PowerPlatform, you can make the rule 0016 more strict.

Not sure if there's a performance penalty, I should consider on getting/reading the setting from the json-file? I think it's a on-time reading the json file if I'm right. Otherwise the sequence of the if-statements can slightly be rearranged.

### CODE REVIEW

File: Design/Rule0016CheckForMissingCaptions.cs [hunk 1]
```diff
@@ -26,6 +27,12 @@ private void CheckForMissingCaptions(SymbolAnalysisContext context)
             if (context.Symbol.IsObsoletePending || context.Symbol.IsObsoleteRemoved) return;
             if (context.Symbol.GetContainingObjectTypeSymbol().IsObsoletePending || context.Symbol.GetContainingObjectTypeSymbol().IsObsoleteRemoved) return;
 
+            LinterSettings.Create();
+            if (!(LinterSettings.instance.enableRule0016ForApiObjects) && context.Symbol.GetContainingObjectTypeSymbol().GetProperty(PropertyKind.PageType) != null)
+            {
+                if (context.Symbol.GetContainingObjectTypeSymbol().GetProperty(PropertyKind.PageType).ValueText == PageTypeKind.API.ToString()) return;
```
Comment (StefanMaron) [inline Design/Rule0016CheckForMissingCaptions.cs:33]: Do you really need to compare them as text here? I think it should be fine without `ValueText` and without `ToString()`

**Review by StefanMaron — CHANGES_REQUESTED** [review]

File: Design/Rule0016CheckForMissingCaptions.cs (hunk 1 above)
Comment (Arthurvdv) [reply Design/Rule0016CheckForMissingCaptions.cs:33]: @StefanMaron, thank you for reviewing this and your feedback.

I'm not sure howto handle the comparison directly between the PageType and the PageTypeKind. Can I directly do and compare between them?

Comment (rvanbekkum) [reply Design/Rule0016CheckForMissingCaptions.cs:33]: I think you could first cast the output from `GetContainingObjectTypeSymbol()` to the page symbol class and then just use the `.PageType` property of that object (which returns a `PageTypeKind` enum value.

I would really love to see this PR merged and released 😊 as this is a very annoying false positive.

Comment (Arthurvdv) [reply Design/Rule0016CheckForMissingCaptions.cs]: Hi @StefanMaron, I've (finally) had the time to rework this. Again, thank you for reviewing this, and you were right that the String/Value wasn't the best solution.

**Review by StefanMaron — APPROVED** [review]

**Pull request merged**